demos/free_surface/explicit_free_surface.py

```diff
@@ -0,0 +1,187 @@
+        self.L = self.D  # Length of the domain in m
+        self.L0 = self.D  # characteristic length scale for scaling the equations
+        ny = self.nx
+        self.left_id, self.right_id, self.bottom_id, self.top_id = 1, 2, 3, 4  # Boundary IDs
+        self.mesh = RectangleMesh(self.nx, ny, self.L/self.L0, self.D/self.L0)  # Rectangle mesh generated via firedrake
+        self.ds = ds  # Need this so that cylindrical case can overload this later with CombinedSurfaceMeasure
+
+    def setup_function_space(self):
+        self.Z = MixedFunctionSpace([self.V, self.W])  # Mixed function space for velocity and pressure.
+
+    def setup_variables(self):
+        # Function to store the solutions:
+        self.z = Function(self.Z)  # a field over the mixed function space Z.
+        self.stokes_vars = self.z.subfunctions
+        self.eta = Function(self.Z[1], name="eta")  # Define a free surface function
+
+        # Next rename for output:
+        self.stokes_vars[0].rename("Velocity")
+        self.stokes_vars[1].rename("Pressure")
+
+    def initialise_wavenumber(self):
+        lam_dimensional = self.D/2  # wavelength of load in m
+        self.lam = lam_dimensional/self.L0  # dimensionless lambda
+        self.kk = Constant(2 * pi / self.lam)  # wavenumber (dimensionless)
+
+    def initialise_temperature(self):
+        self.T = Function(self.Q, name="Temperature").assign(0)  # Setup a dummy function for temperature
+
+    def initialise_approximation(self):
+        Ra = Constant(0)  # Rayleigh number, here we set this to zero as there are no bouyancy terms
+        self.approximation = BoussinesqApproximation(Ra)
+
+    def initialise_free_surfaces(self):
+        self.F0 = Constant(1000 / self.L0)  # initial free surface amplitude (dimensionless)
+        self.eta.interpolate(self.F0 * cos(self.kk * self.X[0]))  # Initial free surface condition
+        self.eta_analytical = Function(self.eta, name="eta analytical")
+
+    def setup_bcs(self):
+        # No normal flow except on the free surface
+        self.stokes_bcs = {
+            self.top_id: {'normal_stress': self.rho0 * self.g * self.eta},  # Apply stress on free surface
+            self.bottom_id: {'un': 0},
+            self.left_id: {'un': 0},
+            self.right_id: {'un': 0},
+        }
+
+    def setup_nullspaces(self):
+        # Nullspaces and near-nullspaces:
+        self.Z_nullspace = None  # Default: don't add nullspace for now
+        self.Z_near_nullspace = None  # Default: don't add nullspace for now
+
+    def setup_solver(self):
+        # Set up the stokes solver
+        self.stokes_solver = StokesSolver(self.z, self.T, self.approximation, bcs=self.stokes_bcs, mu=self.mu, cartesian=self.cartesian,
+                                          free_surface_dt=self.dt, nullspace=self.Z_nullspace, transpose_nullspace=self.Z_nullspace,
+                                          near_nullspace=self.Z_near_nullspace)
```

Comment:
I think I confused myself here and thought that `free_surface_dt` meant the free surface is being handled in `StokesSolver`, but that's only the case where there's *also* a `'free_surface'` boundary condition. And in the implicit case, this method is being overridden anyway.

Reply:
Thanks I actually didn't mean to put this free_surface_dt here because as you say it doesnt actually do anything if there is no bc! (I've also removed the nullsapce because the implict case will overwrite this anyway